### DISCUSSION TITLE
Spherical Coordinate Source Term

### DISCUSSION THREAD

**Discussion by hyluoo** [PrincetonUniversity/athena#562] [Q&A] [answered]
Hi all,
In which file/part of the code can I see how the geometric source is done in spherical coordinate?
In particular, the equation 19 in the Athena++ method paper is confusing for me and I would like to see how it is implemented in practice.
![S_phi](https://github.com/PrincetonUniversity/athena/assets/106158092/59e3fe57-5062-4157-8ef8-7272c4d7c7c4)

**Comment by felker** [comment]
https://github.com/PrincetonUniversity/athena/blob/e58b549877582762bae52ec9bc91b5bd0de43ee3/src/hydro/srcterms/rotating_system_srcterms.cpp#L58-L74

In the future, I recommend searching the source code for formulas using something like `grep -nri "cot("`

**Comment by c-white** [comment]
The above is for a rotating (i.e. noninertial) source term. The base case of the particular discretization of the spherical divergence operator, whether or not the reference frame is rotating, is found in the appropriate coordinates file:

https://github.com/PrincetonUniversity/athena/blob/e58b549877582762bae52ec9bc91b5bd0de43ee3/src/coordinates/spherical_polar.cpp#L442-L526

**Reply by hyluoo** [reply]
Thank you Chris that's exactly what I was looking for

**Reply by hyluoo** [reply, marked as answer]
<img width="780" alt="source4" src="https://github.com/PrincetonUniversity/athena/assets/106158092/62ae231f-1a3b-4482-a4be-48b9f7f0e997">

I checked the code, the source term should be like this, is the one in paper a typo?
